specutils/spectra/spectrum_collection.py

```diff
@@ -58,6 +59,10 @@ def __init__(self, flux, spectral_axis=None, wcs=None, uncertainty=None,
             if not (flux.shape == spectral_axis.shape):
                 raise ValueError("Shape of all data elements must be the same.")
 
+            # Convert to SpectralCoord if simple Quantity was input
+            if not isinstance(spectral_axis, SpectralCoord):
+                spectral_axis = SpectralCoord(spectral_axis)
```

Comment:
```suggestion
            spectral_axis = SpectralCoord(spectral_axis)
```
The ``if`` is unnecessary if #622 is fixed, and in fact this is the primary motivation for a "copy as-is" initializer like this.

So if you think you can roll in a fix to #622 easily here, @rosteen, you might do that and then you can accept the above suggestion.  But as an alternative, maybe just:
```suggestion
            #TODO: change this to just ``spectral_axis = SpectralCoord(spectral_axis)`` when #622 is fixed 
            if not isinstance(spectral_axis, SpectralCoord):
                spectral_axis = SpectralCoord(spectral_axis)
```

Reply:
I'll see if I can roll in a fix for 622 today, don't think it should be difficult

Reply:
@rosteen if you're not able to get #622 implemented, I've also already started (but not finished) an implementation, so feel free to keep it as a separate issue that we can work on later if it's too time-consuming!

Reply:
I have it working, unless I'm missing something more complex that's required. Just coding a bit of test coverage before I push it here.